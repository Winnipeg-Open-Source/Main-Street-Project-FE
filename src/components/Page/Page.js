import React from 'react';
import PropTypes from 'prop-types';
import { Flex, Text } from 'pcln-design-system';

function Page ({ className, title, renderAction, children }) {
    return (
        <Flex className={className} flexDirection='column' width={1} p={3}>
            {title && (
                <Flex justifyContent='space-between' alignItems='center' mb={3}>
                    <Text fontSize={3} bold>{title}</Text>
                    {renderAction()}
                </Flex>
            )}
            {children}
        </Flex>
    );
}

Page.displayName = 'Page';

Page.propTypes = {
    className: PropTypes.string,
    title: PropTypes.string,
    renderAction: PropTypes.node,
    children: PropTypes.node,
};

Page.defaultProps = {
    className: '',
};

export default Page;
