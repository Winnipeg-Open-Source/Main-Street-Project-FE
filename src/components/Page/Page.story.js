import React from 'react';
import Page from './Page';
import { Button, Input } from 'pcln-design-system';

function Children () {
    return (
        <>
            <Input mb={2} />
            <Input />
        </>
    );
}

function RenderAction () {
    return (
        <Button>Render Action Button</Button>
    );
}

export const WithoutTitle = () => (
    <Page>
        <Children />
    </Page>
);

export const WithTitle = () => (
    <Page title='Title'>
        <Children />
    </Page>
);

export const WithActionButton = () => (
    <Page title='Title' renderAction={RenderAction}>
        <Children />
    </Page>
);

export default {
    title: 'Page',
    component: Page,
};
