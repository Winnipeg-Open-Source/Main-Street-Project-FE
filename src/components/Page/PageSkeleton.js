import React, { memo } from 'react';
import PropTypes from 'prop-types';
import Skeleton from 'react-loading-skeleton';
import Page from './Page';

function PageSkeleton ({ className, children }) {
    return (
        <Page
            className={className}
            title={<Skeleton width={200} height={29} />}
            renderAction={() => <Skeleton width={100} height={29} />}
        >
            {children}
        </Page>
    );
}

PageSkeleton.displayName = 'PageSkeleton';

PageSkeleton.propTypes = {
    className: PropTypes.string,
    children: PropTypes.node,
};

PageSkeleton.defaultProps = {
    className: '',
};

export default memo(PageSkeleton);
