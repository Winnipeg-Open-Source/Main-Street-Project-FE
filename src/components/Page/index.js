import WithSkeleton from 'hocs/WithSkeleton';
import Page from './Page';
import PageSkeleton from './PageSkeleton';

export default WithSkeleton(PageSkeleton, Page);
