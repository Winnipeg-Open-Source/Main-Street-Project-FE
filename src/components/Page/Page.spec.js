import React from 'react';
import renderWithTheme from 'tests/helpers/renderWithTheme';
import Page from './Page';

function Children () {
    return (
        <>
            <div>First div</div>
            <div>Second div</div>
        </>
    );
}

function RenderAction () {
    return (
        <div>Render Action</div>
    );
}

describe ('Page', () => {
    it ('renders without crashing', () => {
        const { asFragment } = renderWithTheme(<Page />);

        expect(asFragment()).toMatchSnapshot();
    });

    it ('renders without title', () => {
        const { queryByText } = renderWithTheme(
            <Page>
                <Children />
            </Page>
        );

        expect(queryByText(/First div/i)).toBeInTheDocument();
        expect(queryByText(/Second div/i)).toBeInTheDocument();
        expect(queryByText(/Title/i)).not.toBeInTheDocument();
        expect(queryByText(/Render Action/i)).not.toBeInTheDocument();
    });

    it ('renders with title', () => {
        const { queryByText } = renderWithTheme(
            <Page title='Title'>
                <Children />
            </Page>
        );

        expect(queryByText(/First div/i)).toBeInTheDocument();
        expect(queryByText(/Second div/i)).toBeInTheDocument();
        expect(queryByText(/Title/i)).toBeInTheDocument();
        expect(queryByText(/Render Action/i)).not.toBeInTheDocument();
    });

    it ('renders with render actions', () => {
        const { queryByText } = renderWithTheme(
            <Page title='Title' renderAction={RenderAction}>
                <Children />
            </Page>
        );

        expect(queryByText(/First div/i)).toBeInTheDocument();
        expect(queryByText(/Second div/i)).toBeInTheDocument();
        expect(queryByText(/Title/i)).toBeInTheDocument();
        expect(queryByText(/Render Action/i)).toBeInTheDocument();
    });
});
