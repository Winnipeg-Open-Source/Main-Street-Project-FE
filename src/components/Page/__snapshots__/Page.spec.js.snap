// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-LzLMl bRtbEd sc-LzLMm jaRkvP"
  >
    <div
      class="sc-AykKC sc-AykKE iYWgzp"
      width="1"
    />
  </div>
</DocumentFragment>
`;
